### DISCUSSION TITLE
[solved] Send single MQTT as value and not json

### DISCUSSION THREAD

**Discussion by astrakid** [arendst/Tasmota#21073] [General]
hi,
is it possible to send mqtt-data not in json but as direct value?
reason: i send energy meter (grid) value to mqtt and use it on different purposes / tools. it would be easiest to have it available sa value directly. otherwise i would need to get the value and put it back to mqtt (twice a second).

regards,
andre

**Comment by sfromis** [comment]
If you want such a broken and difficult to handle message format, you can do it yourself using rules to generate extra payloads.

Any decent tool can easily handle JSON data already.

**Reply by astrakid** [reply]
``tried it the last hours. as far as i understood i can't use rules because i am using a script.
i am using precision of my sensor53 with "+16", so each change is transmitted via mqtt. how can i trigger a manual publish within the script?

i tried:
`>D
>B
->sensor53 r
>M 1
+1,3,s,0,9600,HptZhl,hauptzaehler
1,77070100010800ff@1,verbrauch,kWh,bezug,3
1,77070100020800ff@1,einspeisung,kWh,lieferung,3
1,77070100100700ff@1,aktuelleLeistung,W,leistung,16
=>leistung1=HptZhl#leistung
=>publish tasmota_strom/leistung %leistung1%
#`

is that possible?

**Reply by sfromis** [reply]
Yeah, you should be able to execute the `Publish` command via the scripter too. As I'm not really using this, I can't suggest details. I'd not expect it to work as part of the meter definition, and the scripter docs page does have examples using `Publish`.

But again, it should be *trivial* for your tools to deal with JSON.

**Reply by astrakid** [reply]
> But again, it should be trivial for your tools to deal with JSON.

yes, i agree. but currently i have to deail with this at 4 points, and 1 of those 4 is not as easy to implement. so i think it makes sense to do the conversion initially, because it saves the 4 points of doing this conversion.
regards,
andre

**Reply by astrakid** [reply]
1 step ahead: i have to use 
>E
to publish all changes. but i don't get the variable filled with the current value yet. and after a few seconds the device restarts and my script changes are gone.

**Reply by astrakid** [reply]
this is how it works:
`>S`
`=>publish topic/subtopic %sml[3]%`

this simply takes the third decoded value and publishes it as mqtt-topic "subtopic"
